integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java

```diff
@@ -1,77 +1,261 @@
+                Log.then(LOGGER, format(" Environment '%s' create has been done with '%s' KMS key. ", environmentName, encryptionKey));
+            }
+        } else if (CloudPlatform.AZURE.equals(cloudPlatform)) {
+            if (StringUtils.isEmpty(encryptionKey)) {
+                LOGGER.error(format("DES key is not available for '%s' environment!", environmentName));
+                throw new TestFailException(format("DES key is not available for '%s' environment!", environmentName));
+            } else {
+                LOGGER.info(format("Environment '%s' create has been done with '%s' DES key.", environmentName, encryptionKey));
+                Log.then(LOGGER, format(" Environment '%s' create has been done with '%s' DES key. ", environmentName, encryptionKey));
+            }
+        }
+    }
+
+    private FreeIpaTestDto verifyFreeIpaVolumeEncryptionKey(TestContext testContext, FreeIpaTestDto testDto, FreeIpaClient freeIpaClient) {
+        CloudFunctionality cloudFunctionality = testContext.getCloudProvider().getCloudFunctionality();
+        List<String> instanceIds = freeIpaInstanceUtil.getInstanceIds(testDto, freeIpaClient, MASTER.getName());
+        verifyVolumeEncryptionKey(testDto.getCloudPlatform(), testDto.getName(), instanceIds, cloudFunctionality);
+        return testDto;
+    }
+
+    private SdxTestDto verifySdxVolumeEncryptionKey(TestContext testContext, SdxTestDto testDto, SdxClient sdxClient) {
+        CloudFunctionality cloudFunctionality = testContext.getCloudProvider().getCloudFunctionality();
+        List<String> instanceIds = sdxUtil.getInstanceIds(testDto, sdxClient, MASTER.getName());
+        verifyVolumeEncryptionKey(testDto.getCloudPlatform(), testDto.getName(), instanceIds, cloudFunctionality);
+        return testDto;
+    }
+
+    private void verifyVolumeEncryptionKey(CloudPlatform cloudPlatform, String resourceName, List<String> instanceIds,
+            CloudFunctionality cloudFunctionality) {
+        if (CloudPlatform.AWS.equals(cloudPlatform)) {
+            String kmsKeyArn = awsCloudProvider.getEncryptionKeyArn(true);
+            List<String> volumeKmsKeyIds = new ArrayList<>(cloudFunctionality.listVolumeEncryptionKeyIds(resourceName, instanceIds));
+            if (volumeKmsKeyIds.stream().noneMatch(keyId -> keyId.equalsIgnoreCase(kmsKeyArn))) {
+                LOGGER.error(format("Volume has not been encrypted with '%s' KMS key!", kmsKeyArn));
+                throw new TestFailException(format("Volume has not been encrypted with '%s' KMS key!", kmsKeyArn));
+            } else {
+                LOGGER.info(format("Volume has been encrypted with '%s' KMS key.", kmsKeyArn));
+                Log.then(LOGGER, format(" Volume has been encrypted with '%s' KMS key. ", kmsKeyArn));
+            }
+        } else if (CloudPlatform.AZURE.equals(cloudPlatform)) {
+            String desKeyUrl = azureCloudProvider.getEncryptionKeyUrl();
```

Comment:
diskEncryptionSetId (DES) is different from encryptionKeyUrl. Azure creates DES using encryptionKeyURL and uses DES for encryption.
Please can you paste the output of `List<String> volumesDesId = new ArrayList<>(cloudFunctionality.listVolumeEncryptionKeyIds(resourceName, instanceIds))` - I am unsure if it contains DES details.

Reply:
`listVolumeEncryptionKeyIds` shows `Instance 'azure-test-c0e39b3b3015m0' has attached volumes [[/subscriptions/3ddda1c7-d1f5-4e7b-ac81-0523f483b3b3/resourceGroups/azure-test-c0e39b3b3015/providers/Microsoft.Compute/disks/azuretestc0e39b3b-m-0-0-d41d8cd98f00b2]] and disk encryption sets [[/subscriptions/3ddda1c7-d1f5-4e7b-ac81-0523f483b3b3/resourceGroups/AZURE-TEST-2A7DE84E6C2144339-CDP_DES-0C5EA00A-444E-4ADD-9225-E60A8FB1A20C/providers/Microsoft.Compute/diskEncryptionSets/azure-test-2a7de84e6c2144339-DES-0c5ea00a-444e-4add-9225-e60a8fb1a20c]].` and returning `[/subscriptions/3ddda1c7-d1f5-4e7b-ac81-0523f483b3b3/resourceGroups/AZURE-TEST-2A7DE84E6C2144339-CDP_DES-0C5EA00A-444E-4ADD-9225-E60A8FB1A20C/providers/Microsoft.Compute/diskEncryptionSets/azure-test-2a7de84e6c2144339-DES-0c5ea00a-444e-4add-9225-e60a8fb1a20c]`

Reply:
Thank you for pasting the output. My doubt is resolved.